docs/developer-docs/setup/hello-world.mdx

```diff
@@ -0,0 +1,254 @@
+
+For information about project structure and default configuration, [learn more about deploying your first canister](first-canister.mdx).
+ 
+## Smart contract code
+
+<Tabs groupId="language">
+<TabItem value="motoko" label="Motoko" default>
+
+```motoko
+// src/hello_backend/main.mo
+actor {
+public query func greet(name : Text) : async Text {
+    return "Hello, " # name # "!";
+};
+};
+```
+
+</TabItem>
+<TabItem value="rust" label="Rust">
+
+```rust
+// src/hello_world_backend/src/lib.rs
+#[ic_cdk::query]
+fn greet(name: String) -> String {
+    format!("Hello, {}!", name)
+}
+```
+
+</TabItem>
+
+<TabItem value="typescript" label="TypeScript">
+
+```typescript
+// azle_hello/src/index.ts
+import { Canister, query, text, update, Void } from 'azle';
+
+// This is a global variable that is stored on the heap
+let message = '';
+
+export default Canister({
+    // Query calls complete quickly because they do not go through consensus
+    getMessage: query([], text, () => {
+        return message;
+    }),
+    // Update calls take a few seconds to complete
+    // This is because they persist state changes and go through consensus
+    setMessage: update([text], Void, (newMessage) => {
+        message = newMessage; // This change will be persisted
+    })
+});
+```
+
+</TabItem>
+
+<TabItem value="python" label="Python">
+
+Create the project directory and file structure:
+
+```python
+# kybra_hello/src/main.py
+
+from kybra import query, update, void
+
+# This is a global variable that is stored on the heap
+message: str = ''
+
+# Query calls complete quickly because they do not go through consensus
+@query
+def get_message() -> str:
+    return message
+
+# Update calls take a few seconds to complete
+# This is because they persist state changes and go through consensus
+@update
+def set_message(new_message: str) -> void:
+    global message
+    message = new_message # This change will be persisted
+```
+
+</TabItem>
+</Tabs>
+
+
+## Deploy locally 
+
+Deploy the project's canisters locally:
+
+```
+dfx deploy
+```
+
+## Call the canister
+
+<Tabs groupId="language">
+<TabItem value="motoko" label="Motoko" default>
+
+```bash
+dfx canister call hello_backend greet world
+```
+
+Output:
+
+```
+("Hello, world!")
+```
+
+</TabItem>
+<TabItem value="rust" label="Rust">
+
+```bash
+dfx canister call hello_backend greet world
+```
+
+Output:
+
+```
+("Hello, world!")
+```
+
+</TabItem>
+
+<TabItem value="typescript" label="TypeScript">
+
+```bash
+dfx canister call azle_hello setMessage '("Hello, world!")'
+```
+
+Then, call the `getMessage` function:
+
+```
+dfx canister call azle_hello getMessage
+```
+
+Output:
+
+```
+("Hello, world!")
+```
+
+</TabItem>
+
+<TabItem value="python" label="Python">
+
+```bash
+dfx canister call kybra_hello set_message '("Hello, world!")'
+```
+
+Then, call the `getMessage` function:
+
+```
+dfx canister call kybra_hello getMessage
+```
+
+Output:
+
+```
+("Hello, world!")
+```
+
+</TabItem>
+</Tabs>
+
```

Comment:
This command performance many important things behind the scene that warrant explanation:
1. It looked up the address of the smart contract (called canister-id) by it name `hello_backend` in `./.dfx/local/canister_ids.json`.
2. It looked up the interface of the smart contract in `src/declarations/hello_backend/hello_backend.did`.
3. It looked up the signature of function `greet` in the interface.
4. It constructed a message for the `greet` endpoint of the smart context with the given `world` as an argument.
5. It signed the message with the private key of the developer account.
6. It sent the message to the local testnet.
7. Once the message has been executed, it decoded the result using the interface file and printed it.